primitives/src/crh/bowe_hopwood/mod.rs

```diff
@@ -8,23 +8,23 @@ use std::{
 
 use super::pedersen::{PedersenCRH, PedersenWindow};
 use crate::crh::FixedLengthCRH;
-use algebra::{biginteger::BigInteger, fields::PrimeField, groups::Group};
+use algebra::{biginteger::BigInteger, curves::Curve, fields::PrimeField};
 use serde::{Deserialize, Serialize};
 
 pub const CHUNK_SIZE: usize = 3;
 
 #[derive(Clone, Default, Serialize, Deserialize)]
-#[serde(bound(deserialize = "G: Group"))]
-pub struct BoweHopwoodPedersenParameters<G: Group> {
+#[serde(bound(deserialize = "G: Curve"))]
+pub struct BoweHopwoodPedersenParameters<G: Curve> {
```

Comment:
I don't really like the change, here and in other places, of template from G: Group to G: Curve. I guess the only reason is the UniformRand requirement. Since that we introduced it again anyway, I propose to re-introduce it and restore all these templates back as they were before

Reply:
Done